src/input_common/keyboard.cpp

```diff
@@ -12,20 +12,37 @@ namespace InputCommon {
 
 class KeyButton final : public Input::ButtonDevice {
 public:
-    explicit KeyButton(std::shared_ptr<KeyButtonList> key_button_list_)
-        : key_button_list(std::move(key_button_list_)) {}
+    explicit KeyButton(std::shared_ptr<KeyButtonList> key_button_list_, bool toggle_)
+        : key_button_list(std::move(key_button_list_)), toggle(toggle_) {}
 
     ~KeyButton() override;
 
     bool GetStatus() const override {
+        if (toggle) {
+            return toggled_status.load();
+        }
         return status.load();
     }
 
+    void ToggleButton() {
+        if (!lock) {
+            lock = true;
+            toggled_status.store(!toggled_status.load());
+        }
+    }
+
+    void UnlockButton() {
+        lock = false;
+    }
+
     friend class KeyButtonList;
 
 private:
     std::shared_ptr<KeyButtonList> key_button_list;
     std::atomic<bool> status{false};
+    std::atomic<bool> toggled_status{false};
+    bool lock = {};
```

Comment:
```suggestion
    bool lock{false};
```

to match the convention of the variables above it